### ISSUE TITLE
Error 401 Establishing Live Connection

### ISSUE THREAD

**Issue by mag911** [DaxStudio/DaxStudio#769]
DaxStudio launched as an External Tool from PowerBI with Shift held down to create logfile.
PowerBI successfully running with a valid, logged in user account.

28-Nov-21 4:31:58 PM  Establishing Connection
28-Nov-21 4:31:59 PM  The following error occurred while updating the connection: The remote server returned an error: (401) Unauthorized.
28-Nov-21 4:31:59 PM  Error Connecting: Object reference not set to an instance of an object

/* log file */
[DaxStudio-20211128.log](https://github.com/DaxStudio/DaxStudio/files/7613094/DaxStudio-20211128.log)

**Comment by dgosbell** [comment]
So that error would usually mean either the XMLA endpoint is not enabled or the user does not have build permissions on the dataset, both of these conditions need to be true for a 3rd party application like DAX Studio to connect to a data set on the Power BI Service.

**Comment by mag911** [comment]
Ah, well that makes sense then. Apparently the dataset I’ve been given access to is read-only. Cheers. If a messagebox had popped up and said what you’ve just said .. that would’ve been handy ;) Keep up the good work.

**Comment by dgosbell** [comment]
Yes currently I'm just relaying the error that comes back from Power BI, but it's not a very helpful one.

There should be a warning in the output pane which I added in 2.17.3 saying that "Please check your Tenant / PPU admin settings to make sure the XMLA Endpoint is enabled" and I might add "and the current user has at least Build permission or better" to this. It's a bit tricky as it could also be a typo in the username, password or connection string.

But I think you are right that the build permission requirement is not immediately obvious and is likely to catch people out.

**Comment by billzCFGH** [comment]
DAX Studio launched as an External Tool from Power BI with Shift held down to create logfile.
Power BI successfully running with a valid, logged in user account.

02-Feb-23 07:54:39 PM Establishing Connection
02-Feb-23 07:56:29 PM The following error occurred while updating the connection: The remote server returned an error: (401) Unauthorized.
02-Feb-23 07:56:29 PM Error Connecting: **The URI prefix is not recognized**

/* log file */
[DaxStudio-20230215.log](https://github.com/DaxStudio/DaxStudio/files/10744954/DaxStudio-20230215.log)
